docs/v1/api.md

```diff
@@ -0,0 +1,239 @@
+
+```yaml
+kind: ServiceInstanceClaim
+apiVersion: service-catalog.k8s.io/alpha
+metadata:
+  name: my-app-db-claim
+  namespace: my-app
+status: unknown
+spec:
+  serviceClass: postgres-large
+  targetBinding: my-app-binding
+  status: unknown # only written by the service controller
+  statusDescription: unknown # only written by the service controller
+```
+
+## ServiceInstanceBinding - Single Tenant
+
+This example shows the output of a single-tenant `ServiceInstanceClaim`. It is the result of a controller executing a claim that directly references a `ServiceClass`.
+
+```yaml
+kind: ServiceInstanceBinding
+apiVersion: service-catalog.k8s.io/alpha
+metadata:
+  name: my-app-binding # same as the previous 'targetBinding' field
+  namespace: my-app
+status:
+  serviceClass: postgres-large
+  secrets:
+    - my-zip-code-creds-password
+  configMaps:
+    - my-zip-code-creds-username
+    - my-zip-code-creds-conn-info
+```
+
+## ServiceInstanceClaim - Multi Tenant
+
+This example shows a `ServiceInstanceClaim` that references a `ServiceInstance`. The controller that executes this claim will only do a bind.
+
+```yaml
+kind: ServiceInstanceClaim
+apiVersion: service-catalog.k8s.io/alpha
+metadata:
+  name: my-zip-code-db-claim
+  namespace: my-app
+spec:
+  serviceInstance: zip-code-db
+  targetBinding: my-zip-code-binding
+  status: unknown # only written by the service controller
+  statusDescription: unknown # only written by the service controller
+```
+
+## ServiceInstanceBinding - Multi Tenant
+
+This example shows the output of a multi-tenant `ServiceInstanceClaim`. It is the result of a controller executing a claim that just references a `ServiceInstance`.
+
+```yaml
+kind: ServiceInstanceBinding
+apiVersion: service-catalog.k8s.io/alpha
+metadata:
+  name: my-zip-code-binding # same as the above 'targetBinding' field
+  namespace: my-app
+status:
+  serviceInstance: zip-code-db
+  secrets:
+    - my-zip-code-creds-password
+  configMaps:
+    - my-zip-code-creds-username
+    - my-zip-code-creds-conn-info
+```
+
+# The Service Catalog Controller
+
+As indicated above, application developers and cluster operators interact with the aforementioned Kubernetes resources. The service catalog controller watches these resources and takes action on them.
+
+Like ingress controllers, the operator is free to choose any service controller they prefer. All implementations, however, must satisfy the following requirements:
+
+- Watch the event stream for new or deleted `ServiceInstance`s and `ServiceInstanceClaim`s
+- On `ServiceInstance` creation, the controller must do a provision
+- On `ServiceInstance` deletion, the controller must do a deprovision. The controller should not delete any existing `ServiceInstanceClaim`s or `ServiceInstanceBinding`s that reference the deleted `ServiceInstance`
```

Comment:
What happens during modification of the Claims or Bindings with dangling pointers?